hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java

```diff
@@ -59,29 +59,19 @@ public HiveClientPool(int poolSize, Configuration conf) {
   @Override
   protected IMetaStoreClient newClient() {
     try {
-      try {
-        return GET_CLIENT.invoke(
-            hiveConf, (HiveMetaHookLoader) tbl -> null, HiveMetaStoreClient.class.getName());
-      } catch (RuntimeException e) {
-        // any MetaException would be wrapped into RuntimeException during reflection, so let's
-        // double-check type here
-        if (e.getCause() instanceof MetaException) {
```

Comment:
@nk1506 why is this being removed?

Reply:
Fixed it.

Reply:
Let me lookout/add some tests which can catch these errors.

Reply:
@nk1506 this isn't fixed. The logic in the try-catch blocks changed. The only thing that should be changing is adding a null check before calling `t.getMessage()`

Reply:
the other try-catch block looks redundant . We were wrapping into [MetaException](https://github.com/apache/iceberg/blob/main/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java#L69) and throwing and again [wrapping](https://github.com/apache/iceberg/blob/main/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveClientPool.java#L74) into RuntimeMetaException . 

Should I revert and add only null check ?

Reply:
yes, please revert. The scope within the PR should reflect what's being proposed in the commit msg